### PR TITLE
Add SortMeRNA 4.3.3 + FIle cleanup update to SortMeRNA 4.2.0

### PR DESCRIPTION
**Pull request by ffmmulder**
New attempt: Added SortMeRNA 4.3.3 module and added temporary fastq cleanup to SortMeRNA 4.2.0 module

### CODE REVIEW

File: SortMeRNA/4.2.0/SortMeRNA.nf [hunk 1]
```diff
@@ -31,6 +31,8 @@ process SortMeRNA {
             gzip -f < non-rRNA-reads.fastq > ${fastq_files[0].simpleName}_non_rRNA.fastq.gz
             gzip -f < rRNA-reads.fastq > ${fastq_files[0].simpleName}_filtered_rRNA.fastq.gz
             mv rRNA-reads.log ${report_title}_rRNA_report.txt
+            rm non-rRNA-reads.fastq
+            rm rRNA-reads.fastq
```
Comment (rernst) [inline SortMeRNA/4.2.0/SortMeRNA.nf:35]: Would it not be possible to use the `zip-out` option, like you use in 4.3.3? Assuming this results the .gz output.

File: SortMeRNA/4.3.3/SortMeRNA.nf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    tag {"SortMeRNA ${sample_id}"}
+    label 'SortMeRNA_4_3_3'
+    container = 'quay.io/biocontainers/sortmerna:4.3.3--h9ee0642_0'
+    shell = ['/bin/bash', '-euo', 'pipefail']
+    
+    input:
+        tuple(sample_id, rg_id, path(fastq_files))
+        path(db_fasta) 
+    
+
```
Comment (rernst) [inline SortMeRNA/4.3.3/SortMeRNA.nf:11]: Double newline.

File: SortMeRNA/4.3.3/SortMeRNA.nf [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        path(db_fasta) 
+    
+    
+    output:
+        tuple(sample_id, rg_id, path("*_non_rRNA.fastq.gz"), emit: non_rRNA_fastqs)
+        path("*_filtered_rRNA.fastq.gz", emit: rRNA_fastqs)
+        path("*_rRNA_report.txt", emit: qc_report)
+    
+    script:
+        def Refs =  db_fasta.collect{ "$it" }.join(" -ref ")
```
Comment (rernst) [inline SortMeRNA/4.3.3/SortMeRNA.nf:18]: Use lowercase -> `def refs = ....`

File: SortMeRNA/4.3.3/SortMeRNA.nf [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        def report_title = fastq_files[0].simpleName.split("_R1_")[0]  
+        if (params.singleEnd) {
```
Comment (rernst) [inline SortMeRNA/4.3.3/SortMeRNA.nf:20]: Use lowercase -> `params.single_end`

**Review by rernst — CHANGES_REQUESTED** [review]
Just some minor styling changes.

File: SortMeRNA/4.2.0/SortMeRNA.nf (hunk 1 above)
Comment (ffmmulder) [reply SortMeRNA/4.2.0/SortMeRNA.nf:35]: zip-out is only introduced in 4.3.1 (first update in a year since 4.2.0) so not an option unfortunately

**Comment by ffmmulder** [comment]
Regarding the lowercase params and other style issues, will address them... These are also present in the 4.2.0 module (apparently the style wasn't followed completely back then ;) ) so most likely also requires some change in the RNASeq-NF pipeline. To be continued

File: SortMeRNA/4.2.0/SortMeRNA.nf (hunk 1 above)
Comment (rernst) [reply SortMeRNA/4.2.0/SortMeRNA.nf:35]: bummer :D

**Pull request closed without merging**